### PR TITLE
Backport of #735 to secp256k1-sys 0.10

### PR DESCRIPTION
**Pull request by apoelstra**
This backports #735. I am PR'ing to the secp256k1-0.29.x branch because this will work and because it feels like unnecessary complication to try to create a secp256k1-sys-0.10.x branch, which might be "more correct" but would make our git tree harder to understand and maintain.

As in #735, this just deletes some dead code from secp256k1-sys/depend/secp256k1.

### CODE REVIEW

**Comment by Kixunil** [comment]
Do we want to ignore CI?

**Comment by apoelstra** [comment]
Yes. We only just started making an effort to fix CI to pin tooling last week. We don't want to try to backport all those changes.

**Review by Kixunil — APPROVED** [review]
ACK bb026c4909de1058d0cee39e20db1edd1998e6d4

**Comment by apoelstra** [comment]
Forgot to update the lockfiles (which in 0.29.x exist but are not checked in CI). Will get my local CI working on this before merging..

**Review by apoelstra — COMMENTED** [review]
Successfully ran local tests on 21f67d5dcf1afacb73e1f01b83e570c6c258b699.

**Comment by apoelstra** [comment]
Note to self: needed to add 

```
SECP_VENDOR_VERSION_CODE=0_10_0 \
```

to my local CI script to get it to pass, since by default it wants the symbols in secp-sys to match the version in Cargo.toml exactly. (But doing this would turn a non-breaking change into a breaking one, which is why I didn't.)

**Comment by Kixunil** [comment]
@apoelstra your CI fail made me realize we should do the same thing here as we do in `bitcoinconsensus`.

**Comment by apoelstra** [comment]
What in particular?

**Comment by Kixunil** [comment]
Version the API, not the library and put the upstream version into metadata instead.

**Comment by apoelstra** [comment]
Agreed. Though for purposes of this PR, we can't make such a change (as it'd be a breaking change to change our version format at all).

**Comment by Kixunil** [comment]
Well, we already are breaking our format here, but obviously I'm not NACKing this.

**Comment by apoelstra** [comment]
>obviously I'm not NACKing this

Will you ACK it? :}

Fine if you don't want to, but there aren't a lot of maintainers on this repo and you're the one who appears to be active right now.

**Comment by Kixunil** [comment]
I already did and since GH didn't dismiss my review I haven't noticed the force push.

**Comment by apoelstra** [comment]
Tagged and published.

**Pull request merged**
